glusterd2/commands/volumes/profile-create.go

```diff
@@ -0,0 +1,57 @@
+package volumecommands
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/store"
+	"github.com/gluster/glusterd2/pkg/api"
+	"github.com/gluster/glusterd2/pkg/errors"
+)
+
+func volumeProfileCreateHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	var req api.ProfileCreateReq
+	if err := restutils.UnmarshalRequest(r, &req); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusUnprocessableEntity, errors.ErrJSONParsingFailed.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	resp, err := store.Store.Get(context.TODO(), "groupoptions")
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	var groupOptions map[string][]api.Option
+	if err := json.Unmarshal(resp.Kvs[0].Value, &groupOptions); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	if _, ok := groupOptions[req.ProfileName]; ok {
```

Comment:
A user should be allowed to replace an existing grouped option.

And an API to delete too ? You can disallow deleting default/built-in grouped option.